eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/DbsnpJsonClusteredVariantsWriter.java

```diff
@@ -51,16 +49,18 @@ public DbsnpJsonClusteredVariantsWriter(MongoTemplate mongoTemplate, ImportCount
     @Override
     public void write(List<? extends DbsnpClusteredVariantEntity> clusteredVariants) {
         try {
-            dbsnpClusteredVariantWriter.write(clusteredVariants);
+            if (!clusteredVariants.isEmpty()) {
+                dbsnpClusteredVariantWriter.write(clusteredVariants);
+            }
         } catch (BulkOperationException exception) {
-            List<String> ids = extractUniqueHashesForDuplicateKeyError(exception).collect(toList());
-            Query query = new Query();
-            query.addCriteria(Criteria.where("_id").in(ids));
-            List<DbsnpClusteredVariantEntity> entities = mongoTemplate.find(query, DbsnpClusteredVariantEntity.class);
-            List<Long> rsIDs = entities.stream()
-                .map(DbsnpClusteredVariantEntity::getAccession)
-                .collect(toList());
-            logger.error("Duplicate RS IDs: {}", rsIDs);
+            List<String> hashes = extractUniqueHashesForDuplicateKeyError(exception).collect(toList());
```

Comment:
if you are only going to test presence, use a set.
```suggestion
            Set<String> hashes = extractUniqueHashesForDuplicateKeyError(exception).collect(toSet());
```

Reply:
Doesn't the **extractUniqueHashesForDuplicateKeyError** method return only unique hashes?

Reply:
yes but I think this is one of the slowest parts of the pipeline, and if there are lots of collisions, using a set will be faster. probably O(n) instead of O(n*n) where n is the batch size